discovery/ec2/ec2.go

```diff
@@ -107,6 +114,11 @@ func (c *SDConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
 		}
 		c.Region = region
 	}
+	for _, f := range c.Filters {
```

Comment:
You should add a config test for this, they're under the config/testdata directory.